file_optical.c

```diff
@@ -1670,13 +1670,68 @@ static int fbo_handle_cmd(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 	return 0;
 }
 
+static int fbo_update_cfgstring(struct tcmu_device *dev,
+				struct tcmulib_cfg_info *cfg)
+{
+	char *oldptr;
+	int ret;
+
+	if (!tcmu_check_state(dev, TCMU_DEV_STATE_CAPACITY_CHANGE_FOR_CONFIG_CHANGE)) {
+		tcmu_dev_err(dev, "You need to include size change to be able to change dev_config\n");
+		return -EINVAL;
+	}
+	tcmu_remove_state(dev, TCMU_DEV_STATE_CAPACITY_CHANGE_FOR_CONFIG_CHANGE);
+
+	tcmu_update_state(dev, TCMU_DEV_STATE_CONFIG_CHANGING);
+
+	fbo_close(dev);
+
+	oldptr = cfg->data.dev_cfgstring;
+	tcmu_set_dev_cfgstring(dev, oldptr);
+
+	ret = fbo_open(dev);
+	if (ret != 0) {
+		tcmu_dev_err(dev, "Failed opening of new media\n");
+		goto fail_open;
+	}
+
+	tcmu_remove_state(dev, TCMU_DEV_STATE_CONFIG_CHANGING);
```

Comment:
I mean it is possible, so a possible fix would be to move the update state so that is before the remove state. Then if a cmd does come in then the check for TCMU_DEV_STATE_CONFIG_CHANGED would be executed.